google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/TableDependencyTracker.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate;
+
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import org.hibernate.boot.Metadata;
+import org.hibernate.mapping.Table;
+
+/**
+ * Tracks the order in which tables should be processed (created/dropped) by Hibernate.
+ *
+ * <p>In Spanner, interleaved tables must be created or dropped in the correct order. For example,
+ * one must create a parent table before its interleaved table. And one must drop the interleaved
+ * table first before its parent table.
+ */
+public class TableDependencyTracker {
```

Comment:
Maybe it should package-private so it doesn't become part of the public API?

Reply:
Done!